usr.sbin/ctld/ctld.h

```diff
@@ -49,25 +57,34 @@
 #define	MAX_LUNS			1024
 #define	SOCKBUF_SIZE			1048576
 
+struct isns_req;
+struct port;
+
 struct auth {
-	TAILQ_ENTRY(auth)		a_next;
-	struct auth_group		*a_auth_group;
-	char				*a_user;
-	char				*a_secret;
-	char				*a_mutual_user;
-	char				*a_mutual_secret;
-};
+	auth(const char *secret) : a_secret(secret) {}
+	auth(const char *secret, const char *mutual_user,
+	    const char *mutual_secret) : a_secret(secret),
+	    a_mutual_user(mutual_user), a_mutual_secret(mutual_secret) {}
+
+	bool mutual() const { return !a_mutual_user.empty(); }
 
-struct auth_name {
-	TAILQ_ENTRY(auth_name)		an_next;
-	struct auth_group		*an_auth_group;
-	char				*an_initiator_name;
+	const char *secret() const { return a_secret.c_str(); }
```

Comment:
More examples of single-line vs do we split them with new lines after the return type, before the opening brace, before the statement, and before the closing brace, e.g.:

```
          const char *
          secret() const
          {
                   return (a_secret.c_str());
          }

          ...
```